irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -275,24 +280,87 @@ DROP TABLE IF EXISTS index_by_id_height_asset;
       storage->committed = true;
     }
 
+    template <typename Query>
+    class StorageImpl::Deleter {
```

Comment:
maybe make it just a standalone class, and put it in cpp only, since its not used anywhere else

Also add comment about the purpose of this class

Reply:
Also to remove "friendliness" of this class you can pass connection_ directly